docker/docker-entrypoint.sh

```diff
@@ -1,28 +1,39 @@
 #!/bin/bash
 
 # Parameters
-if [[ "${NAT_IP_AUTO}" == "true" && -z "${NAT_PUBLIC_IP_AUTO}" ]]; then
-  export CODEX_NAT=$(hostname --ip-address)
-  echo "Internal: Set CODEX_NAT: ${CODEX_NAT}"
+if [[ -z "${CODEX_NAT}" ]]; then
+  if [[ "${NAT_IP_AUTO}" == "true" && -z "${NAT_PUBLIC_IP_AUTO}" ]]; then
```

Comment:
OK so the semantics here is that if you set NAT_PUBLIC_IP_AUTO that will override NAT_IP_AUTO, correct? I like it.

Reply:
Yes, and this is done for dist-tests images, where we set [`nat_ip_auto: true`](https://github.com/codex-storage/nim-codex/blob/master/.github/workflows/docker-dist-tests.yml#L28) by default and we need a way to set Public IP if needed.